miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt

```diff
@@ -143,11 +140,11 @@ abstract class MiniAppMessageBridge {
             ActionType.SHARE_INFO.action -> onShareContent(callbackObj.id, jsonStr)
             ActionType.LOAD_AD.action -> adBridgeDispatcher.onLoadAd(callbackObj.id, jsonStr)
             ActionType.SHOW_AD.action -> adBridgeDispatcher.onShowAd(callbackObj.id, jsonStr)
-            ActionType.GET_USER_NAME.action -> userInfoBridgeDispatcher.onGetUserName(callbackObj.id)
```

Comment:
I think, if the purpose is to reduce the size of the class, then we can skip changing anything in `MiniAppMessageBridge` which will be safer for the HostApp also.

Rather we can improve your idea e.g.

```
internal class UserInfoBridgeWrapper(
    private val bridgeExecutor: MiniAppBridgeExecutor,
    private val customPermissionCache: MiniAppCustomPermissionCache,
    private val miniAppId: String,
    private val userInfoBridgeDispatcher: UserInfoBridgeDispatcher
) {

    fun onGetUserName(callbackId: String) {
        // move the implementation from `UserInfoBridgeDispatcher` to here like you already did.
    }
}

// then you can call it inside `UserInfoBridgeDispatcher`.

abstract class UserInfoBridgeDispatcher {

      internal fun init(
        bridgeExecutor: MiniAppBridgeExecutor,
        miniAppCustomPermissionCache: MiniAppCustomPermissionCache,
        miniAppId: String
       ) {
        ...
        this.userInfoWrapper = UserInfoBridgeWrapper(bridgeExecutor, customPermissionCache, miniAppId, this)
       }
   
    internal fun onGetUserName(callbackId: String) = userInfoWrapper.onGetUserName(callbackId)
}
```

What do you think?

Reply:
This one has a cross reference problem. Anyway, not only `UserInfoBridgeDispatcher`, the `MiniAppMessageBridge` is smaller and not need to maintain the complexity of initialization as before.

Reply:
Yeah, +1 for removing complexity of initialization from `MiniAppMessageBridge`.